ampd-handlers/src/bin/evm/handler.rs

```diff
@@ -0,0 +1,456 @@
+                        },
+                    );
+                })
+                .collect();
+            info!(
+                votes = votes.as_value(),
+                "ready to vote for messages in poll"
+            );
+
+            votes
+        });
+
+        Ok(vec![self
+            .vote_msg(poll_id, votes)
+            .into_any()
+            .expect("vote msg should serialize")])
+    }
+
+    fn subscription_params(&self) -> SubscriptionParams {
+        SubscriptionParams::new(
+            vec![AbciEventTypeFilter {
+                event_type: PollStartedEvent::event_type(),
+                contract: self.voting_verifier_contract.clone(),
+            }], // TODO: Add verifier set poll started event filter?
+            false,
+        )
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::convert::TryInto;
+
+    use ampd::evm::finalizer::Finalization;
+    use ampd::evm::json_rpc::MockEthereumClient;
+    use ampd::handlers::test_utils::{into_structured_event, participants};
+    use ampd::monitoring;
+    use ampd::types::{Hash, TMAddress};
+    use ampd_sdk::event::event_handler::EventHandler;
+    use ampd_sdk::grpc::client::test_utils::MockHandlerTaskClient;
+    use axelar_wasm_std::chain_name;
+    use axelar_wasm_std::msg_id::HexTxHashAndEventIndex;
+    use axelar_wasm_std::voting::Vote;
+    use error_stack::{Report, Result};
+    use ethers_core::types::{Block, H160, H256, U64};
+    use ethers_providers::ProviderError;
+    use events::Error::{DeserializationFailed, EventTypeMismatch};
+    use events::Event;
+    use tokio::test as async_test;
+    use voting_verifier::events::{PollMetadata, PollStarted, TxEventConfirmation};
+
+    use super::PollStartedEvent;
+    use crate::handler::Handler;
+
+    const PREFIX: &str = "axelar";
+    const ETHEREUM: &str = "ethereum";
+
+    fn poll_started_event(participants: Vec<TMAddress>, expires_at: u64) -> PollStarted {
+        let msg_ids = [
+            HexTxHashAndEventIndex::new(H256::repeat_byte(1), 0u64),
+            HexTxHashAndEventIndex::new(H256::repeat_byte(2), 1u64),
+            HexTxHashAndEventIndex::new(H256::repeat_byte(3), 10u64),
+        ];
+        PollStarted::Messages {
+            metadata: PollMetadata {
+                poll_id: "100".parse().unwrap(),
+                source_chain: chain_name!(ETHEREUM),
+                source_gateway_address: "0x4f4495243837681061c4743b74eedf548d5686a5"
+                    .parse()
+                    .unwrap(),
+                confirmation_height: 15,
+                expires_at,
+                participants: participants
+                    .into_iter()
+                    .map(|addr| cosmwasm_std::Addr::unchecked(addr.to_string()))
+                    .collect(),
+            },
+            #[allow(deprecated)] // TODO: The below events use the deprecated tx_id and event_index fields. Remove this attribute when those fields are removed
```

Comment:
why are they using deprecated fields?

Reply:
they are not, but looks like the message and `allow` was left there. I'll remove it